corehq/apps/app_manager/detail_screen.py

```diff
@@ -102,12 +102,13 @@ def __init__(self, app, module, detail, column, sort_element=None,
 
     @property
     def locale_id(self):
-        if not is_sort_only_column(self.column):
+        if is_sort_only_column(self.column):
+            sort_element = [sort_element for sort_element in self.detail.sort_elements if sort_element.field == self.column.field][0]
```

Comment:
If you add the display text to the `SortOnlyDetailColumn` when it get's created you should just be able to use here instead of finding having to find the actual sort element. You'd also have to make `sort_only_column_header_locale` method not require the actual sort element.